multik-api/src/main/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Add.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.kotlinx.multik.ndarray.operations
+
+import org.jetbrains.kotlinx.multik.ndarray.data.*
+
+public fun <T : Number> D1Array<T>.append(n: D1Array<T>): D1Array<T>{
+    val totalSize :Int = this.size+n.size
+    when (this.dtype) {
+        DataType.IntDataType -> {
+            var dest = IntArray(totalSize)
+            System.arraycopy(this.data.getIntArray(),0,dest,0,this.size)
+            System.arraycopy(n.data.getIntArray(), 0, dest,this.size, n.size)
+            var appendedArr:D1Array<T> = D1Array(data = initMemoryView<T>(totalSize, this.dtype){dest[it] as T},dim=D1, shape = intArrayOf(totalSize), dtype = this.dtype)
```

Comment:
when we know the type of the array, we can immediately create `MemoryView` - `MemoryViewIntArray(dest)`, so we will wrap the corresponding array, without copying

Reply:
Hey @devcrocod , changing the view to be specific type would also return that D1Array of that specific type instead of D1Array< T >. Altho it could be type casted as such 
```kt
val appendedArr: D1Array<Int> = D1Array(data =MemoryViewIntArray(dest),dim=D1, shape = intArrayOf(totalSize), dtype = this.dtype)
appendedArr as D1Array<T>
```
should i just type cast it that ways then?

Reply:
here the problem is that we are going through the array once again and copying.
https://github.com/Kotlin/multik/blob/cd26d4e7904ae9e9a574b64467ef7a93dc5d1beb/multik-api/src/main/kotlin/org/jetbrains/kotlinx/multik/ndarray/data/MemoryView.kt#L726